src/lib/storage/segment_iterables/any_segment_iterator.hpp

```diff
@@ -116,19 +116,27 @@ class AnySegmentIterator : public BaseSegmentIterator<AnySegmentIterator<T>, Seg
 
  public:
   AnySegmentIterator(const AnySegmentIterator& other) : _wrapper{other._wrapper->clone()} {}
+  AnySegmentIterator& operator=(const AnySegmentIterator& other) {
+    _wrapper = other._wrapper->clone();
```

Comment:
```suggestion
    if (this == &other) { return *this; }
    _wrapper = other._wrapper->clone();
```

This page says that the object should perform no action on self-assignment: https://en.cppreference.com/w/cpp/language/operators

Reply:
Nice. Self-assignment is something that I never really gave any attention to.
https://isocpp.org/wiki/faq/assignment-operators#self-assignment-how